src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -1268,73 +1269,117 @@ def test_flow_with_nan_inf_metrics(self, pf: PFClient, monkeypatch) -> None:
+                id="with_async_func",
+            ),
+        ],
+    )
+    def test_flex_flow_run(
+        self,
+        pf,
+        run_params: dict,
+        expected_snapshot_yaml: dict,
+        extra_check: Callable[[], bool],
+        expected_details: dict,
+    ) -> None:
+        run = pf.run(**run_params)
         assert run.status == "Completed"
         assert "error" not in run._to_dict()
-        # will create a YAML in run snapshot
-        local_storage = LocalStorageOperations(run=run)
-        assert local_storage._dag_path.exists()
-        # the YAML file will not exist in user's folder
-        assert not Path(f"{EAGER_FLOWS_DIR}/simple_without_yaml/flow.flex.yaml").exists()
 
-    def test_eager_flow_yaml_override(self, pf):
-        run = pf.run(
-            flow="entry2:my_flow2",
-            code=f"{EAGER_FLOWS_DIR}/multiple_entries",
-            data=f"{DATAS_DIR}/simple_eager_flow_data.jsonl",
-        )
-        assert run.status == "Completed"
-        assert "error" not in run._to_dict()
+        assert extra_check()
+
         # will create a YAML in run snapshot
         local_storage = LocalStorageOperations(run=run)
         assert local_storage._dag_path.exists()
-        # original YAMl content not changed
-        original_dict = load_yaml(f"{EAGER_FLOWS_DIR}/multiple_entries/flow.flex.yaml")
-        assert original_dict["entry"] == "entry1:my_flow1"
-
-        # actual result will be entry2:my_flow2
-        details = pf.get_details(run.name)
-        # convert DataFrame to dict
-        details_dict = details.to_dict(orient="list")
-        assert details_dict == {"inputs.line_number": [0], "outputs.output": ["entry2flow2"]}
-
-    def test_flex_flow_with_func(self, pf):
-        run = pf.run(
-            flow=my_entry,
-            data=f"{DATAS_DIR}/simple_eager_flow_data.jsonl",
-            # set code folder to avoid snapshot too big
-            code=f"{EAGER_FLOWS_DIR}/multiple_entries",
-            column_mapping={"input1": "${data.input_val}"},
-        )
-        assert run.status == "Completed"
-        assert "error" not in run._to_dict()
 
-        # actual result will be entry2:my_flow2
-        details = pf.get_details(run.name)
-        # convert DataFrame to dict
-        details_dict = details.to_dict(orient="list")
-        assert details_dict == {"inputs.input1": ["input1"], "inputs.line_number": [0], "outputs.output": ["input1"]}
-
-        run = pf.run(
-            flow=my_async_entry,
-            data=f"{DATAS_DIR}/simple_eager_flow_data.jsonl",
-            # set code folder to avoid snapshot too big
-            code=f"{EAGER_FLOWS_DIR}/multiple_entries",
-            column_mapping={"input2": "${data.input_val}"},
-        )
-        assert run.status == "Completed"
-        assert "error" not in run._to_dict()
+        yaml_dict = load_yaml(local_storage._dag_path)
+        assert yaml_dict == expected_snapshot_yaml
```

Comment:
Add check for at least one non-flex run?

Reply:
this is flex only feature?